roles/sap_ha_install_hana_hsr/tasks/configure_hsr.yml

```diff
@@ -1,61 +1,68 @@
   when:
-    - sap_ha_install_hana_hsr_role == 'secondary'
+    - ansible_hostname == item.node_name
+    - item.node_role is defined
+    - item.node_role == 'primary'
+    - ansible_hostname not in __sap_ha_install_hana_hsr_srstate.stdout
+  loop: "{{ sap_ha_install_hana_hsr_cluster_nodes }}"
+  loop_control:
+    label: "{{ item.node_name }}"
 
-# [2] Register secondary node to HANA System Replication
-- name: SAP HSR - [2] Register secondary node to HANA System Replication
+- name: "SAP HSR - Register secondary node to HANA System Replication"
   shell: |
-      source /usr/sap/{{ sap_ha_install_hana_hsr_sid }}/home/.sapenv.sh && \
-      /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/hdbnsutil \
-      -sr_register --name={{ sap_ha_install_hana_hsr_alias }} \
-      --remoteHost={{ sap_ha_install_hana_hsr_primary_hostname }} --remoteInstance={{ sap_ha_install_hana_hsr_instance_number }} \
-      --replicationMode={{ sap_ha_install_hana_hsr_rep_mode }} --operationMode={{ sap_ha_install_hana_hsr_oper_mode }} \
-      --online
+    source /usr/sap/{{ sap_ha_install_hana_hsr_sid }}/home/.sapenv.sh && \
+    /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/hdbnsutil \
+    -sr_register --name={{ item.hana_site }} \
+    --remoteHost={{ __sap_ha_install_hana_hsr_primary_node }} --remoteInstance={{ sap_ha_install_hana_hsr_instance_number }} \
+    --replicationMode={{ sap_ha_install_hana_hsr_rep_mode }} --operationMode={{ sap_ha_install_hana_hsr_oper_mode }} \
+    --online
   args:
     executable: /bin/bash
   become: true
   become_user: "{{ sap_ha_install_hana_hsr_sid | lower }}adm"
   register: registersr
-  changed_when: "'adding site' in registersr.stdout"
   when:
-    - sap_ha_install_hana_hsr_role == 'secondary'
+    - ansible_hostname == item.node_name
+    - item.node_role is not defined or item.node_role == 'secondary'
+    - ansible_hostname not in __sap_ha_install_hana_hsr_srstate.stdout
+  loop: "{{ sap_ha_install_hana_hsr_cluster_nodes }}"
+  loop_control:
+    label: "{{ item.node_name }}"
 
-# [2] Start HANA instance
-- name: SAP HSR - [2] Start HANA instance
+- name: "SAP HSR - Start HANA instance on secondary"
   shell: |
-      /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/sapcontrol \
-      -nr {{ sap_ha_install_hana_hsr_instance_number }} -function StartSystem
+    /usr/sap/{{ sap_ha_install_hana_hsr_sid | upper }}/HDB{{ sap_ha_install_hana_hsr_instance_number }}/exe/sapcontrol \
+    -nr {{ sap_ha_install_hana_hsr_instance_number }} -function StartSystem
   args:
     executable: /bin/bash
   become: true
   become_user: "{{ sap_ha_install_hana_hsr_sid | lower }}adm"
   register: startinstance
   changed_when: "'StartSystem' in startinstance.stdout"
   when:
```

Comment:
Why these conditions are not similar to conditions for primary node? Shouldn't they be the same, but secondary instead of primary?

Reply:
The dictionary key "node_role" could be left undefined for secondary nodes. The idea was to save repeated "node_role: secondary" definitions for any of potentially many nodes that are not the one primary node.

Reply:
@kksat I am now resolving this conversation after @ja9fuchs replied and explained it. I also see that you approved the PR already. Feel free to open an issue related your comment if you have further questions.